src/pkg/boltz.ts

```diff
@@ -281,6 +305,68 @@ export class Boltz implements BoltzInterface {
     return Extractor.extract(finalizer.pset);
   }
 
+  getRefundTransaction({
+    utxo,
+    refundKeyPair,
+    redeemScript,
+    destinationScript,
+    fee,
+    blindingPublicKey,
+  }: GetRefundTransactionParams): Transaction {
+    if (!utxo.blindingData) throw new Error('utxo is not blinded');
+    if (!utxo.witnessUtxo) throw new Error('utxo missing witnessUtxo');
+    const pset = Creator.newPset();
+    const updater = new Updater(pset);
+
+    updater
+      .addInputs([
+        {
+          txid: utxo.txid,
+          txIndex: utxo.vout,
+          witnessUtxo: utxo.witnessUtxo,
+          sighashType: Transaction.SIGHASH_ALL,
+        },
+      ])
+      .addInWitnessScript(0, redeemScript)
+      .addOutputs([
+        {
+          script: destinationScript,
+          blindingPublicKey,
+          asset: this.asset,
+          amount: (utxo.blindingData?.value ?? 0) - fee,
+          blinderIndex: 0,
+        },
+        {
+          amount: fee,
+          asset: this.asset,
+        },
+      ]);
+
+    const blindedPset = this.blindPset(pset, {
+      index: 0,
+      value: utxo.blindingData.value.toString(),
+      valueBlindingFactor: Buffer.from(utxo.blindingData.valueBlindingFactor, 'hex'),
+      asset: AssetHash.fromHex(utxo.blindingData.asset).bytesWithoutPrefix,
+      assetBlindingFactor: Buffer.from(utxo.blindingData.assetBlindingFactor, 'hex'),
+    });
+
+    const signedPset = this.signPset(blindedPset, refundKeyPair);
+
+    const finalizer = new Finalizer(signedPset);
+
+    finalizer.finalizeInput(0, (inputIndex, pset) => {
+      return {
+        finalScriptSig: undefined,
+        finalScriptWitness: witnessStackToScriptWitness([
+          pset.inputs[inputIndex].partialSigs![0].signature,
+          Buffer.of(), //dummy preimage
```

Comment:
The only difference between a refund and claim tx, is that you can pass an empty preimage

https://github.com/BoltzExchange/boltz-core/blob/master/lib/liquid/swap/Refund.ts#L6

Reply:
false: you also need the heightLocktime to be set on the input :)